addons/product/models/product_packaging.py

```diff
@@ -64,3 +70,44 @@ def _find_suitable_product_packaging(self, product_qty, uom_id):
             if new_qty == product_qty:
                 return packaging
         return self.env['product.packaging']
+
+    @api.model
+    def _compute_packaging(self, product_packaging_id, product_id, product_uom_qty, product_uom, optional_type=False):
+        """Helper method to be used to compute packaging and packaging_qty for
+        stock.move, purchase.order.line, and sale.order.line.
+        If there is product_packaging_id, check with current product_uom_qty, the
+        packaging_qty is an integer or not. If no, remove it.
+        If there is no product_packaging_id or it's removed during the above process,
+        try suggest a packaging with integer packaging_qty.
+        """
+        if not product_id or not product_uom_qty or not product_uom:
+            return False
```

Comment:
`product_packaging` is also expected, I prefer to do the same like for `_compute_packaging_qty`, so that to make those two methods stay together. What do you think?

Reply:
It's expected but not always present, whereas the product is.
Anyway, it's already a good simplification to drop the `product_packaging_id` param and use self instead :+1: